src/python/sortingview/gui/extensions/unitstable/Units/TableWidget.tsx

```diff
@@ -188,7 +206,8 @@ const TableWidget: FunctionComponent<Props> = (props) => {
                         }
                         setSortFieldOrder(newSortFieldOrder)
                     }}
-                    onDeselectAll={selectedRowIds.length > 0 ? () => {onSelectedRowIdsChanged([])} : undefined}
+                    onDeselectAll={selectedRowIds.length === sortedRows.length ? handleDeselectAll : undefined}
```

Comment:
Show select-all checkbox if not everything is selected. Show deselect-all checkbox if everything is selected. So, to deselect all, you must first select all.